File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        # feelslike : merge intevals
        # 
        #           ____
        #    ___
        #  ___          _____

        # at x point how many ballons can we burst 
        # 

        n=len(points)
        if n == 1:
            return 1
        p=sorted(points)

        cnt=1
        x=p[0][0]
        y=p[0][1]


        for i in range(1,n,1):
            if(y >= p[i][0]):
                x=max(x,p[i][0])
                y=min(y,p[i][1])
            else :
                cnt+=1
                x=p[i][0]
                y=p[i][1]
        

        return cnt

        

File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {

        int n = points.size();
        if(n==1)return 1;
        sort(points.begin(),points.end());
    
        int cnt=1;
        int x=points[0][0];
        int y=points[0][1];

        for(int i=1;i<n;i++){
            if(points[i][0]<=y){
                x=max(x,points[i][0]);
                y=min(y,points[i][1]);
            }
            else{
                cnt++;
                x=points[i][0];
                y=points[i][1];
            }

            // cout<<x<<" "<<y<<endl;
        }

        return cnt;

        
    }
};